src/primitive-field/primitive-field.component.ts

```diff
@@ -71,12 +72,10 @@ export class PrimitiveFieldComponent extends AbstractFieldComponent implements O
   }
 
   commitValueChange() {
-    // Validation
     let errors = this.schemaValidationService.validateValue(this.value, this.schema);
-    if (!errors.length) {
-      this.jsonStoreService.setIn(this.path, this.value);
-    }
-    this.errors = errors;
+    this.jsonStoreService.setIn(this.path, this.value);
+    this.internalErrors = errors;
+    this.appGlobalsService.internalErrors = {[this.pathString]: errors};
```

Comment:
So instead of creating that new object `{[this.pathString]: errors};` and assigning it, you could create a new class that extends [Map](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map) and override the `set` method with custom behaviour.